src/clojure/dialog/logger.clj

```diff
@@ -133,15 +141,21 @@
 (defn get-level
   "Get the current level setting for a logger. If no logger name is provided,
   this returns the root logger's level."
-  ([]
+  (^:deprecated []
    (or (:level config) :info))
   ([logger]
-   (or (get @level-cache logger)
-       (let [level (or (match-block logger)
-                       (match-level logger)
-                       (get-level))]
-         (swap! level-cache assoc logger level)
-         level))))
+   (get-level {:level (:level config)
+               :levels (:levels config)
+               :blocked (:blocked config)
+               :level-cache level-cache} logger))
+  ([options logger]
+   (let [{:keys [level blocked levels level-cache]} options]
+     (or (get @level-cache logger)
+         (let [level (or (match-block blocked logger)
+                         (match-level levels logger)
+                         level)]
+           (swap! level-cache assoc logger level)
```

Comment:
Hmm, thinking through this a bit further - having a cache in the outputs is problematic, because it's not going to be cleared when someone calls `set-level!` the way the global cache is. So if your configured level for `acme.foo` is `:error`, and your output sees a log message from `acme.foo.bar` for the first time, your output cache will now have `{"acme.foo.bar" :error}`. Then an operator wants to drop the level to `:debug` to investigate some behavior, your output will still think the level is `:error` and won't emit those messages.

Reply:
I don't think I completely follow but I trust your judgement.